src/lib/utils/chunk_pruning_utils.hpp

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include <set>
+
+#include <boost/functional/hash.hpp>
+
+#include "types.hpp"
+
+namespace hyrise {
+
+class StoredTableNode;
+class TableStatistics;
+struct OperatorScanPredicate;
+class PredicateNode;
+
+using PredicatePruningChain = std::vector<std::shared_ptr<PredicateNode>>;
+
+using StoredTableNodePredicateNodePair = std::pair<std::shared_ptr<StoredTableNode>, std::shared_ptr<PredicateNode>>;
+
+std::set<ChunkID> compute_chunk_exclude_list(
+    const PredicatePruningChain& predicate_pruning_chain, const std::shared_ptr<StoredTableNode>& stored_table_node,
+    std::unordered_map<StoredTableNodePredicateNodePair, std::set<ChunkID>,
+                       boost::hash<StoredTableNodePredicateNodePair>>& excluded_chunk_ids_by_predicate);
+
+// Convenience version when no caching of results is required, e.g., when there is only a single PredicatePruningChain.
```

Comment:
Sound like there is no caching, but you just don't require passing it, right?

Reply:
Yes.
```
Convenience version when no the result cache is not used by the caller, e.g., when there is only a single PredicatePruningChain.
```

Reply:
Sounds good.